unit-tests/src/test/java/org/eclipse/collections/impl/bag/sorted/mutable/SynchronizedSortedBagTest.java

```diff
@@ -101,4 +107,34 @@ public void bottomOccurrences()
         Assert.assertEquals("five", bottom5.getLast().getOne());
         Assert.assertEquals(5, bottom5.getLast().getTwo());
     }
+
+    @Override
+    @Test
+    public void collectWithOccurrences()
+    {
+        Bag<Integer> bag = this.newWith(3, 3, 3, 2, 2, 1);
+        Bag<ObjectIntPair<Integer>> actual1 =
+                bag.collectWithOccurrences(PrimitiveTuples::pair, Bags.mutable.empty());
+        Assert.assertEquals(
+                Bags.immutable.with(
+                        PrimitiveTuples.pair(Integer.valueOf(3), 3),
+                        PrimitiveTuples.pair(Integer.valueOf(2), 2),
+                        PrimitiveTuples.pair(Integer.valueOf(1), 1)),
+                actual1);
+        Assert.assertEquals(
+                Lists.mutable.with(
+                        PrimitiveTuples.pair(Integer.valueOf(1), 1),
+                        PrimitiveTuples.pair(Integer.valueOf(2), 2),
+                        PrimitiveTuples.pair(Integer.valueOf(3), 3)),
+                bag.collectWithOccurrences(PrimitiveTuples::pair));
+
+        Set<ObjectIntPair<Integer>> actual2 =
+                bag.collectWithOccurrences(PrimitiveTuples::pair, Sets.mutable.empty());
+        Assert.assertEquals(
+                Sets.immutable.with(
+                        PrimitiveTuples.pair(Integer.valueOf(3), 3),
+                        PrimitiveTuples.pair(Integer.valueOf(2), 2),
+                        PrimitiveTuples.pair(Integer.valueOf(1), 1)),
+                actual2);
```

Comment:
It would be interesting to add an assertion where many input elements transform to the same output element.

Reply:
Done!